homeworks/valitothattila/4_exceptions_logging/to_do.py

```diff
@@ -0,0 +1,103 @@
+import os
+import logging
+
+print(os.getcwd())
+
+file_path = "python_hu/homeworks/valitothattila/4_exceptions_logging/to_do.txt"
+
+
+# ------------------------ Logolás beállítása -------------------------
+file_handler = logging.FileHandler("python_hu/homeworks/valitothattila/4_exceptions_logging/app.log")
+stream_handler = logging.StreamHandler()
+
+formatter = logging.Formatter('%(asctime)s - %(levelname)s - %(message)s')
+
+file_handler.setFormatter(formatter)
+stream_handler.setFormatter(formatter)
+
+file_handler.setLevel(logging.DEBUG)
+stream_handler.setLevel(logging.DEBUG)
+
+logger = logging.getLogger()
+logger.addHandler(file_handler)
+logger.addHandler(stream_handler)
+logger.setLevel(logging.DEBUG)
+
+# ------------------------ Függvények -------------------------
+
+# ------------------------ Menü = 1 -> task_list elem hozzáadása  -------------------------
+def add_to_tasklist(add_element):
+    with open(file_path, "a") as file:
+        file.write(f"{add_element}\n")
+        logger.info(f"Add elemnt to the task list: {add_element}.")
+
+# ------------------------ Menü = 2 -> task_list elemeinek listázása -------------------------
+def list_tasklist():
+    with open(file_path, "r") as file:
+        file_element = file.readlines()
+        index_number = 1
+        for list_element in file_element:
+            print(f"{index_number}: {list_element.strip()}")
+            index_number += 1
+        logger.info("List task elements from the file.")      
+
+# ------------------------ Menü = 3 -> task_list elem törlése -------------------------
+def del_element_of_list(del_index):
```

Comment:
Ez egy picit túl van bonyolítva, de alapvetően az elképzelésed működik, így ezt rádbízom hogy megváltoztatod e (figyelembe véve a pénteki határidőt):
A jelenlegi megoldásod az, hogy direktben olvasol a file-ból és printeled ki a tartalmát a `list_tasklist()`-tel, illetve a direktben törölsz a file-ból.

Mi lenne akkor, ha egyszer beolvasnád a file tartalmát egy listába, és az egész programod ezt a listát használná hozzáadásra és törlésre is? A végén pedig amikor kilép a felhasználó, akkor kiírnád egyszer, egy helyen a listát egy .txt file-ba.
Ehhez ugye az a function ami beolvassa a file-t returnöl vagy listát a beolvasott sorokkal, vagy egy üres listát ha nincs benne adat, vagy egyáltalán nem létezik a file. Ezzel szignifikánsan egyszerűsíteni tudnád a kódod.